src/lib/routes/admin-api/addon.ts

```diff
@@ -44,108 +44,96 @@ class AddonController extends Controller {
             method: 'get',
             path: `${PATH}:id`,
             handler: this.getAddon,
             permission: NONE,
-            middleware: [
-                openApiService.validPath({
-                    summary: 'Get a specific addon',
-                    description:
-                        'Retrieve information about the addon whose ID matches the ID in the request URL.',
-                    tags: ['Addons'],
-                    operationId: 'getAddon',
-                    responses: {
-                        200: createResponseSchema('addonSchema'),
-                        ...getStandardResponses(401),
-                    },
-                }),
-            ],
+            openApi: {
+                summary: 'Get a specific addon',
+                description:
+                    'Retrieve information about the addon whose ID matches the ID in the request URL.',
+                tags: ['Addons'],
+                operationId: 'getAddon',
+                responses: {
+                    200: createResponseSchema('addonSchema'),
+                },
+            },
         });
 
-        this.route({
+        this.routeWithOpenApi(this.openApiService)({
             method: 'put',
             path: `${PATH}:id`,
             handler: this.updateAddon,
             permission: UPDATE_ADDON,
-            middleware: [
-                openApiService.validPath({
-                    summary: 'Update an addon',
-                    description: `Update the addon with a specific ID. Any fields in the update object will be updated. Properties that are not included in the update object will not be affected. To empty a property, pass \`null\` as that property's value.
+            openApi: {
+                summary: 'Update an addon',
+                description: `Update the addon with a specific ID. Any fields in the update object will be updated. Properties that are not included in the update object will not be affected. To empty a property, pass \`null\` as that property's value.
 
 Note: passing \`null\` as a value for the description property will set it to an empty string.`,
-                    tags: ['Addons'],
-                    operationId: 'updateAddon',
-                    requestBody: createRequestSchema('addonCreateUpdateSchema'),
-                    responses: {
-                        200: createResponseSchema('addonSchema'),
-                        ...getStandardResponses(400, 401, 403, 404, 413, 415),
-                    },
-                }),
-            ],
+                tags: ['Addons'],
+                operationId: 'updateAddon',
+                requestBody: createRequestSchema('addonCreateUpdateSchema'),
+                responses: {
+                    200: createResponseSchema('addonSchema'),
+                    ...getStandardResponses(404),
+                },
+            },
         });
 
-        this.route({
+        this.routeWithOpenApi(this.openApiService)({
```

Comment:
maybe openApiService can be passed with the super() call in the constructor so that we don't have to repeat it in each route

Reply:
Yeah, that's definitely also an option! I think I'm personally leaning towards making the class abstract and requiring subclasses to implement it, but it's definitely an option we can explore!